src/webgpu/shader/validation/expression/call/builtin/const_override_validation.ts

```diff
@@ -202,58 +203,66 @@ var<private> v = ${builtin}(${callArgs.join(', ')});`,
  * @param t the ShaderValidationTest
  * @param binaryOp the symbol of the binary operator
  * @param expectedResult false if an error is expected, true if no error is expected
+ * @param leftStage the evaluation stage for the left argument
  * @param left the left-hand side of the binary operation
+ * @param rightStage the evaluation stage for the right argument
  * @param right the right-hand side of the binary operation
- * @param stage the evaluation stage
  */
 export function validateConstOrOverrideBinaryOpEval(
   t: ShaderValidationTest,
   binaryOp: string,
   expectedResult: boolean,
+  leftStage: ExecutionStage,
   left: Value,
-  right: Value,
-  stage: ConstantOrOverrideStage
+  rightStage: ExecutionStage,
+  right: Value
 ) {
   const allArgs = [left, right];
   const elTys = allArgs.map(arg => elementTypeOf(arg.type)!);
   const enables = elTys.some(ty => ty === Type.f16) ? 'enable f16;' : '';
 
-  switch (stage) {
-    case 'constant': {
-      t.expectCompileResult(
-        expectedResult,
-        `${enables}
-const v = ${left.wgsl()} ${binaryOp} ${right.wgsl()};`
-      );
-      break;
-    }
-    case 'override': {
-      assert(!elTys.some(ty => isAbstractType(ty)));
-      const constants: Record<string, number> = {};
-      const overrideDecls: string[] = [];
-      const opArgs: string[] = [];
-      let numOverrides = 0;
-      for (const arg of allArgs) {
+  const codeLines = [enables];
+  const constants: Record<string, number> = {};
+  let numOverrides = 0;
+
+  function addOperand(name: string, stage: ExecutionStage, value: Value) {
+    switch (stage) {
+      case 'runtime':
+        codeLines.push(`var<private> ${name} : ${value.type};`);
```

Comment:
Shouldn't this be initialized with `= ${value.wgsl()}` like the others? We want to make sure that implementations don't try to incorrectly constant-fold through the `var`.